packages/notifications/push/amplify_push_notifications/ios/Classes/AmplifyPushNotificationsPlugin+AppDelegate.swift

```diff
@@ -0,0 +1,89 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import Foundation
+import Flutter
+
+private let completionHandlerIdKey = "completionHandlerId"
+
+extension AmplifyPushNotificationsPlugin {
```

Comment:
Is it a valid filename and way of doing it? `AmplifyPushNotificationsPlugin+AppDelegate.swift` 
Would it still allow apps to have their own AppDelegate?

Reply:
This naming indicate this file contains additional `AppDelegate` behaviors added to `AmplifyPushNotificationsPlugin`. 

Here we implement `FlutterPlugin` protocol defined App life cycle methods, it doesn't block developers to do anything with their own `AppDelegate`.